### PR TITLE
DOC: Capitalize docstring summaries

### LINKED ISSUE #23881: DOC: Capitalize docstring summaries
Our standard is to start the docstrings with a capital letter (and finish them with a period).

We'll shortly add a check to make sure that new contributions follow this standard. But before we can do that, we need to fix the many cases that are already wrong.

In this issue we should fix all the remaining cases where the docstring summary starts with a lower case letter. This is the command to generate them (which should return nothing after the fix), and the list of current issues.
```
$ ./scripts/validate_docstrings.py --errors=SS02
pandas.io.json.json_normalize: Summary does not start with a capital letter
pandas.HDFStore.info: Summary does not start with a capital letter
pandas.melt: Summary does not start with a capital letter
pandas.Series.argmin: Summary does not start with a capital letter
pandas.Series.argmax: Summary does not start with a capital letter
pandas.DataFrame.melt: Summary does not start with a capital letter
pandas.DataFrame.to_feather: Summary does not start with a capital letter
pandas.Index.values: Summary does not start with a capital letter
pandas.Index.is_monotonic: Summary does not start with a capital letter
pandas.Index.is_monotonic_increasing: Summary does not start with a capital letter
pandas.Index.is_monotonic_decreasing: Summary does not start with a capital letter
pandas.Index.is_unique: Summary does not start with a capital letter
pandas.Index.dtype: Summary does not start with a capital letter
pandas.Index.dtype_str: Summary does not start with a capital letter
pandas.Index.inferred_type: Summary does not start with a capital letter
pandas.Index.where: Summary does not start with a capital letter
pandas.Index.take: Summary does not start with a capital letter
pandas.Index.putmask: Summary does not start with a capital letter
pandas.Index.ravel: Summary does not start with a capital letter
pandas.Index.join: Summary does not start with a capital letter
pandas.Index.asof_locs: Summary does not start with a capital letter
pandas.Index.contains: Summary does not start with a capital letter
pandas.Index.get_indexer_for: Summary does not start with a capital letter
pandas.RangeIndex.from_range: Summary does not start with a capital letter
pandas.MultiIndex.to_hierarchical: Summary does not start with a capital letter
pandas.Timestamp.ceil: Summary does not start with a capital letter
pandas.Timestamp.combine: Summary does not start with a capital letter
pandas.Timestamp.floor: Summary does not start with a capital letter
pandas.Timestamp.fromordinal: Summary does not start with a capital letter
pandas.Timestamp.fromtimestamp: Summary does not start with a capital letter
pandas.Timestamp.replace: Summary does not start with a capital letter
pandas.Timestamp.strftime: Summary does not start with a capital letter
pandas.Timestamp.strptime: Summary does not start with a capital letter
pandas.Timedelta.view: Summary does not start with a capital letter
pandas.Timedelta.ceil: Summary does not start with a capital letter
pandas.Timedelta.floor: Summary does not start with a capital letter
pandas.Timedelta.to_pytimedelta: Summary does not start with a capital letter
pandas.tseries.offsets.Tick: Summary does not start with a capital letter
pandas.tseries.offsets.Day: Summary does not start with a capital letter
pandas.tseries.offsets.Hour: Summary does not start with a capital letter
pandas.tseries.offsets.Minute: Summary does not start with a capital letter
pandas.tseries.offsets.Second: Summary does not start with a capital letter
pandas.tseries.offsets.Milli: Summary does not start with a capital letter
pandas.tseries.offsets.Micro: Summary does not start with a capital letter
pandas.tseries.offsets.Nano: Summary does not start with a capital letter
pandas.core.window.Rolling.max: Summary does not start with a capital letter
pandas.core.window.Rolling.cov: Summary does not start with a capital letter
pandas.core.window.Rolling.apply: Summary does not start with a capital letter
pandas.core.window.Rolling.quantile: Summary does not start with a capital letter
pandas.core.window.Expanding.max: Summary does not start with a capital letter
pandas.core.window.Expanding.cov: Summary does not start with a capital letter
pandas.core.window.Expanding.apply: Summary does not start with a capital letter
pandas.core.window.Expanding.quantile: Summary does not start with a capital letter
pandas.core.window.EWM.mean: Summary does not start with a capital letter
pandas.core.window.EWM.std: Summary does not start with a capital letter
pandas.core.window.EWM.var: Summary does not start with a capital letter
pandas.core.window.EWM.corr: Summary does not start with a capital letter
pandas.core.window.EWM.cov: Summary does not start with a capital letter
pandas.core.groupby.GroupBy.groups: Summary does not start with a capital letter
pandas.core.groupby.GroupBy.indices: Summary does not start with a capital letter
pandas.core.resample.Resampler.groups: Summary does not start with a capital letter
pandas.core.resample.Resampler.indices: Summary does not start with a capital letter
pandas.core.resample.Resampler.asfreq: Summary does not start with a capital letter
pandas.io.formats.style.Styler.clear: Summary does not start with a capital letter
pandas.Panel.items: Summary does not start with a capital letter
pandas.Panel.major_axis: Summary does not start with a capital letter
pandas.Panel.minor_axis: Summary does not start with a capital letter
```

Ignore `Series.argmin` and `Series.argmax`, as the docstring is autogenerated, and it's not trivial to fix. And other similar cases, if any.

In some cases the problem is that there is a deprecated directive `.. deprecated::` at the beginning of te docstring. This is wrong too, and the deprecation should be after the short summary (the first line/paragraph of the summary).

While this issue is mainly to make the first letter in the summary a capital letter, for the cases where we fix this, would be nice to also add a period at the end of the paragraph, if there is not got one already.

### PR DESCRIPTION
**Pull request by varadgunjal**
- [x] closes #23881
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] Solves most of the issues raised by the `validate_docstrings.py` script. 

The few issues that still show up on running the script can be attributed to- 
- Docstrings non-existent viz. `tseries.offsets.Tick`, `tseries.offsets.Day` etc. (working on this for a separate PR)
- Auto-generated Doctrings viz. `Series.argmin`, `Timestamp.strptime` etc.
- Docstrings starting with quotes viz. 'Normalizes', 'Unpivots' which were (presumably) added in by the original authors by for a reason.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -5839,7 +5839,7 @@ def unstack(self, level=-1, fill_value=None):
         return unstack(self, level, fill_value)
 
     _shared_docs['melt'] = ("""
-    "Unpivots" a DataFrame from wide format to long format, optionally
+    'Unpivots' a DataFrame from wide format to long format, optionally
```
Comment (datapythonista) [inline pandas/core/frame.py]: I think we can simply remove the quotes. But in any case, we should fix the validation script to not report cases where the first character is not a letter.

File: pandas/core/groupby/base.py [hunk 2]
```diff
@@ -12,11 +12,11 @@
 
 
 class GroupByMixin(object):
-    """ provide the groupby facilities to the mixed object """
+    """ Provide the groupby facilities to the mixed object. """
```
Comment (datapythonista) [inline pandas/core/groupby/base.py]: Do you mind changing all the cases like this (quotes in the same line) to this:
`
"""
Provide the groupby facilities to the mixed object.
"""
`
This will fail the validation soon too, so if you don't mind changing the edited ones, it'd be great.

File: pandas/core/groupby/base.py [hunk 3]
```diff
@@ -28,8 +28,8 @@ def f(x):
 
     def _gotitem(self, key, ndim, subset=None):
         """
-        sub-classes to define
-        return a sliced object
+        Sub-classes to define
+        return a sliced object.
```
Comment (datapythonista) [inline pandas/core/groupby/base.py]: I think they can go to the same line: `Sub-classes to define. Return a sliced object.`

File: pandas/core/groupby/groupby.py [hunk 4]
```diff
@@ -377,14 +377,14 @@ def __unicode__(self):
...
-        sub-classes may have a different policy
+        We create the grouper on instantiation
+        sub-classes may have a different policy.
         """
         pass
 
     @property
     def groups(self):
-        """ dict {group name -> group labels} """
+        """ Dict {group name -> group labels} """
```
Comment (datapythonista) [inline pandas/core/groupby/groupby.py]: Can you finish with period (besides having the text in a line between the quotes, as explained earlier).

File: pandas/core/groupby/groupby.py [hunk 5]
```diff
@@ -395,14 +395,14 @@ def ngroups(self):
 
     @property
     def indices(self):
-        """ dict {group name -> group indices} """
+        """ Dict {group name -> group indices} """
```
Comment (datapythonista) [inline pandas/core/groupby/groupby.py]: Same, missing period.

File: pandas/core/groupby/groupby.py [hunk 6]
```diff
@@ -450,7 +450,7 @@ def get_converter(s):
         return [self.indices.get(name, []) for name in names]
 
     def _get_index(self, name):
-        """ safe get index, translate keys for datelike to underlying repr """
+        """ Safe get index, translate keys for datelike to underlying repr ."""
```
Comment (datapythonista) [inline pandas/core/groupby/groupby.py]: Don't leave the space before the period please.

File: pandas/core/groupby/groupby.py [hunk 7]
```diff
@@ -1127,7 +1127,7 @@ def mean(self, *args, **kwargs):
     @Appender(_doc_template)
     def median(self, **kwargs):
         """
-        Compute median of groups, excluding missing values
+        Compute median of groups, excluding missing values.
 
         For multiple groupings, the result index will be a MultiIndex
```
Comment (datapythonista) [inline pandas/core/groupby/groupby.py]: Do you mind adding a period here too?

File: pandas/core/indexes/base.py [hunk 8]
```diff
@@ -1493,13 +1493,14 @@ def _mpl_repr(self):
     # introspection
     @property
     def is_monotonic(self):
-        """ alias for is_monotonic_increasing (deprecated) """
+        """ Alias for is_monotonic_increasing.
+        .. deprecated :: """
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Besides leaving the quotes in their own lines, do you mind adding a blank line between the sentences

File: pandas/core/indexes/base.py [hunk 9]
```diff
@@ -2536,12 +2537,17 @@ def asof(self, label):
 
     def asof_locs(self, where, mask):
         """
-        where : array of timestamps
-        mask : array of booleans where data is not NA
+        Parameters
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: If you know what this function does, can you add a short summary?

File: pandas/core/indexes/base.py [hunk 10]
```diff
@@ -4281,9 +4288,9 @@ def _maybe_cast_indexer(self, key):
 
     def _validate_indexer(self, form, key, kind):
         """
-        if we are positional indexer
+        If we are positional indexer
         validate that we have appropriate typed bounds
-        must be an integer
+        must be an integer.
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: If you don't mind making the lines go until close to 79 characters, I think this fits in two lines.

File: pandas/core/indexes/base.py [hunk 11]
```diff
@@ -4504,7 +4511,7 @@ def delete(self, loc):
     def insert(self, loc, item):
         """
         Make new Index inserting new item at location. Follows
-        Python list.append semantics for negative values
+        Python list.append semantics for negative values.
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Do you mind adding the second sentence to another paragraph (with one blank line between the first).

File: pandas/core/window.py [hunk 12]
```diff
@@ -995,7 +995,7 @@ def sum(self, *args, **kwargs):
         return self._apply('roll_sum', 'sum', **kwargs)
 
     _shared_docs['max'] = dedent("""
-    %(name)s maximum
+    Calculate the %(name)s maximum
```
Comment (datapythonista) [inline pandas/core/window.py]: Can you add a period at the end?

File: pandas/core/window.py [hunk 13]
```diff
@@ -1333,7 +1333,7 @@ def f(arg, *args, **kwargs):
                            **kwargs)
 
     _shared_docs['cov'] = dedent("""
-    %(name)s sample covariance
+    Calculate the %(name)s sample covariance
```
Comment (datapythonista) [inline pandas/core/window.py]: Can you add a period at the end?

File: pandas/io/formats/style.py [hunk 14]
```diff
@@ -504,7 +505,8 @@ def __deepcopy__(self, memo):
         return self._copy(deepcopy=True)
 
     def clear(self):
-        """"Reset" the styler, removing any previously applied styles.
+        """
+        'Reset' the styler, removing any previously applied styles.
```
Comment (datapythonista) [inline pandas/io/formats/style.py]: I'd remove the quotes here too.

File: pandas/io/json/normalize.py [hunk 15]
```diff
@@ -99,7 +99,7 @@ def json_normalize(data, record_path=None, meta=None,
                    errors='raise',
                    sep='.'):
     """
-    "Normalize" semi-structured JSON data into a flat table
+    'Normalize' semi-structured JSON data into a flat table.
```
Comment (datapythonista) [inline pandas/io/json/normalize.py]: Remove the quotes here too.

**Review by datapythonista — COMMENTED** [review]
Looks great, thanks for working on this. I added some comments with things that besides the capitalization are wrong in the format. If you don't mind taking care of those too, that would be really great.

File: pandas/core/frame.py (hunk 1 above)
Comment (varadgunjal) [reply pandas/core/frame.py]: So, which fix should I make? Removing the quotes or fixing the validation script? Or both?

File: pandas/core/groupby/base.py (hunk 2 above)
Comment (varadgunjal) [reply pandas/core/groupby/base.py]: Sorry, the formatting is slightly unclear - it shows no change from what I've already added in. Did you mean to say I should move it to a new line?

File: pandas/core/groupby/groupby.py (hunk 4 above)
Comment (varadgunjal) [reply pandas/core/groupby/groupby.py]: I actually had a question about this. My `awk` script did this one automatically, but this looks more like a description of what is being returned (I'm not familiar enough with the codebase to be sure of it). Is that the case or should I just go ahead and treat it as the docstring summary?

File: pandas/core/frame.py (hunk 1 above)
Comment (datapythonista) [reply pandas/core/frame.py]: in this PR remove the quotes, if you want to change the script, that should be im a separate PR

File: pandas/core/groupby/base.py (hunk 2 above)
Comment (datapythonista) [reply pandas/core/groupby/base.py]: sorry, I did a mistake with the fithib formatting, amd wasn't rendering as I intended. I editted the comment, you can see now what I meant

File: pandas/core/groupby/groupby.py (hunk 4 above)
Comment (datapythonista) [reply pandas/core/groupby/groupby.py]: this is the summary. In a different PR we can edit it and make the description look less cryptic, but for now jus add the period at the end, so the validation script does not complain

**Comment by varadgunjal** [comment]
I've been diving into the code further to see where I can contribute and have noticed that a lot of these issues are repeated in other files that do not show up when the validate script runs. The changes will affect **a lot** of files. Shall go ahead and make the changes there and submit PRs module-by-module? Or should I just stick to the files that the script found for this issue?

File: pandas/core/groupby/groupby.py (hunk 4 above)
Comment (varadgunjal) [reply pandas/core/groupby/groupby.py]: :+1:

File: pandas/core/indexes/base.py (hunk 9 above)
Comment (varadgunjal) [reply pandas/core/indexes/base.py]: @datapythonista I've added in a summary based on what I've understood from playing around with the function. I think you / the original author of the function should verify it though.

File: pandas/core/indexes/base.py [hunk 16]
```diff
@@ -1493,13 +1524,16 @@ def _mpl_repr(self):
     # introspection
     @property
     def is_monotonic(self):
-        """ alias for is_monotonic_increasing (deprecated) """
+        """
+        Alias for is_monotonic_increasing.
+
+        .. deprecated :: """
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: This is not actually deprecated (see that the code doesn't raise a deprecation warning.

Can you remove the deprecated directive (also the deprecated in brackets as you did).

@jreback do we want to deprecate this? (that would be for another PR)

File: pandas/core/indexes/base.py [hunk 17]
```diff
@@ -1532,8 +1566,8 @@ def is_monotonic_decreasing(self):
 
     @property
     def _is_strictly_monotonic_increasing(self):
-        """return if the index is strictly monotonic increasing
-        (only increasing) values
+        """Return if the index is strictly monotonic increasing
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Can you move this one to the next line of the quotes too?

File: pandas/core/indexes/base.py (hunk 9 above)
Comment (datapythonista) [reply pandas/core/indexes/base.py]: @jreback, do you mind reviewing the summary?

File: pandas/core/indexes/base.py [hunk 18]
```diff
@@ -2536,12 +2588,36 @@ def asof(self, label):
...
+        ----------
+        where : Index
+            An Index consisting of an array of timestamps.
+        mask : array-like
+            Array of booleans denoting where values in the original
+            data are not NA.
+
+        Returns
+        -------
+        result: array
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]:
```suggestion
        numpy.ndarray
```
We used to have a name for what is being returned, but unless more than one thing is returned, we just say the type now

File: pandas/core/indexes/base.py [hunk 19]
```diff
@@ -3191,7 +3267,7 @@ def get_value(self, series, key):
     def set_value(self, arr, key, value):
         """
         Fast lookup of value from 1-dimensional ndarray. Only use this if you
-        know what you're doing
+        know what you're doing.
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Do you mind moving the `Only use this...` to a `Notes` section

File: pandas/core/indexes/base.py [hunk 20]
```diff
@@ -3681,9 +3761,9 @@ def isin(self, values, level=None):
 
     def _can_reindex(self, indexer):
         """
-        *this is an internal non-public method*
+        Check if we are allowing reindexing with this particular indexer.
 
-        Check if we are allowing reindexing with this particular indexer
+        *this is an internal non-public method*
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: You can remove this comment, should be obvious from the starting underscore

File: pandas/core/indexes/base.py [hunk 21]
```diff
@@ -3758,10 +3839,10 @@ def reindex(self, target, method=None, level=None, limit=None,
 
     def _reindex_non_unique(self, target):
         """
-        *this is an internal non-public method*
-
         Create a new index with target's values (move/add/delete values as
-        necessary) use with non-unique Index and a possibly non-unique target
+        necessary) use with non-unique Index and a possibly non-unique target.
+
+        *this is an internal non-public method*
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: same, you can remove this

File: pandas/core/indexes/base.py [hunk 22]
```diff
@@ -3818,11 +3899,11 @@ def _reindex_non_unique(self, target):
         return new_index, indexer, new_indexer
 
     _index_shared_docs['join'] = """
-        *this is an internal non-public method*
-
         Compute join_index and indexers to conform data
         structures to the new index.
 
+        *This is an internal non-public method*
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Not sure if this is actually true (didn't check where this is applied), but I'd remove this too.

File: pandas/core/indexes/base.py [hunk 23]
```diff
@@ -4035,16 +4116,18 @@ def _join_level(self, other, level, how='left', return_indexers=False,
         """
         The join method *only* affects the level of the resulting
         MultiIndex. Otherwise it just exactly aligns the Index data to the
-        labels of the level in the MultiIndex. If `keep_order` == True, the
-        order of the data indexed by the MultiIndex will not be changed;
-        otherwise, it will tie out with `other`.
+        labels of the level in the MultiIndex.
+
+        If `keep_order` == True, the order of the data indexed by the
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]:
```suggestion
        If ```keep_order == True```, the order of the data indexed by the
```

File: pandas/core/indexes/base.py [hunk 24]
```diff
@@ -4802,23 +4891,26 @@ def _add_numeric_methods_disabled(cls):
...
 
     def _validate_for_numeric_binop(self, other, op):
         """
-        return valid other, evaluate or raise TypeError
-        if we are not of the appropriate type
+        Return valid other; evaluate or raise TypeError if we are not of
+        the appropriate type.
 
-        internal method called by ops
+        *this is an internal method called by ops*
```
Comment (datapythonista) [inline pandas/core/indexes/base.py]: Can you add this in a `Notes` section instead of the stars? I guess saying the it's call by ops is useful in this case.

File: pandas/core/indexes/datetimelike.py [hunk 25]
```diff
@@ -34,7 +34,7 @@
 
 
 class DatelikeOps(object):
-    """ common ops for DatetimeIndex/PeriodIndex, but not TimedeltaIndex """
+    """ Common ops for DatetimeIndex/PeriodIndex, but not TimedeltaIndex. """
```
Comment (datapythonista) [inline pandas/core/indexes/datetimelike.py]: Can you move to the next line of the quotes, as you did before.

File: pandas/core/indexes/datetimelike.py [hunk 26]
```diff
@@ -76,7 +76,7 @@ def strftime(self, date_format):
 
 
 class TimelikeOps(object):
-    """ common ops for TimedeltaIndex/DatetimeIndex, but not PeriodIndex """
+    """ Common ops for TimedeltaIndex/DatetimeIndex, but not PeriodIndex. """
```
Comment (datapythonista) [inline pandas/core/indexes/datetimelike.py]: same

File: pandas/core/panel.py [hunk 27]
```diff
@@ -1115,7 +1116,7 @@ def _apply_1d(self, func, axis):
         return self._construct_return_type(results, planes)
 
     def _apply_2d(self, func, axis):
-        """ handle 2-d slices, equiv to iterating over the other axis """
+        """ Handle 2-d slices, equiv to iterating over the other axis. """
```
Comment (datapythonista) [inline pandas/core/panel.py]: if you can also move to the next line of the quotes.

File: pandas/core/panel.py [hunk 28]
```diff
@@ -1172,7 +1173,7 @@ def _reduce(self, op, name, axis=0, skipna=True, numeric_only=None,
         return self._construct_return_type(result, axes)
 
     def _construct_return_type(self, result, axes=None):
-        """ return the type for the ndim of the result """
+        """ Return the type for the ndim of the result. """
```
Comment (datapythonista) [inline pandas/core/panel.py]: same

File: pandas/core/panel.py [hunk 29]
```diff
@@ -1440,13 +1441,13 @@ def _get_join_index(self, other, how):
     # miscellaneous data creation
     @staticmethod
     def _extract_axes(self, data, axes, **kwargs):
-        """ return a list of the axis indices """
+        """ Return a list of the axis indices. """
```
Comment (datapythonista) [inline pandas/core/panel.py]: same

File: pandas/core/panel.py [hunk 30]
```diff
@@ -1440,13 +1441,13 @@ def _get_join_index(self, other, how):
...
         return [self._extract_axis(self, data, axis=i, **kwargs)
                 for i, a in enumerate(axes)]
 
     @staticmethod
     def _extract_axes_for_slice(self, axes):
-        """ return the slice dictionary for these axes """
+        """ Return the slice dictionary for these axes. """
```
Comment (datapythonista) [inline pandas/core/panel.py]: same

File: pandas/core/resample.py [hunk 31]
```diff
@@ -757,8 +757,8 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
 
     def asfreq(self, fill_value=None):
         """
-        return the values at the new freq,
-        essentially a reindex
+        Return the values at the new freq,
+        essentially a reindex.
```
Comment (datapythonista) [inline pandas/core/resample.py]: does it fit in one line

File: pandas/core/resample.py [hunk 32]
```diff
@@ -909,7 +909,7 @@ def _maybe_process_deprecations(r, how=None, fill_method=None, limit=None):
 
 
 class _GroupByMixin(GroupByMixin):
-    """ provide the groupby facilities """
+    """ Provide the groupby facilities. """
```
Comment (datapythonista) [inline pandas/core/resample.py]: separate summary from the quotes lines

File: pandas/core/resample.py [hunk 33]
```diff
@@ -1218,7 +1218,7 @@ def _constructor(self):
 
 
 def resample(obj, kind=None, **kwds):
-    """ create a TimeGrouper and return our resampler """
+    """ Create a TimeGrouper and return our resampler. """
```
Comment (datapythonista) [inline pandas/core/resample.py]: same

File: pandas/core/frame.py [hunk 34]
```diff
@@ -6045,7 +6045,7 @@ def _gotitem(self,
                  ):
         # type: (...) -> Union[Series, DataFrame]
         """
-        sub-classes to define
+        Sub-classes to define
         return a sliced object
```
Comment (datapythonista) [inline pandas/core/frame.py]: Can you use `Sub-classes to define. Return a sliced object.`

Or something better if you want.

**Review by datapythonista — COMMENTED** [review]
Really nice changes, thanks @varadgunjal.

Added some comments, if you can take care of them, and I'll try to get this merged soon, before it causes conflicts (being so big it's likely to happen). Not so important, but for the next time, it usually makes things easier and more efficient if you split a PR of this size in two or three chunks.

File: pandas/core/indexes/base.py (hunk 18 above)
Comment (varadgunjal) [reply pandas/core/indexes/base.py]: I see. It should be noted then that this convention is not followed everywhere in the code. I see the name still listed in many places.

**Comment by varadgunjal** [comment]
Sorry that the PR has become quite big. I've stuck to fixing issues in the files reported by the `validate_docstrings` script. As discussed previously, I will push corresponding changes to docstrings in all the various other files if and when I push changes to the code therein in the future.

File: pandas/core/indexes/base.py (hunk 18 above)
Comment (datapythonista) [reply pandas/core/indexes/base.py]: I know, there is a PR to detect those cases, and after it's merged I'll open an issue so all them are fixed.

File: pandas/tseries/offsets.py [hunk 35]
```diff
@@ -1142,9 +1173,9 @@ class SemiMonthEnd(SemiMonthOffset):
 
     Parameters
     ----------
-    n : int
-    normalize : bool, default False
-    day_of_month : int, {1, 3,...,27}, default 15
+        n : int
+        normalize : bool, default False
+        day_of_month : int, {1, 3,...,27}, default 15
```
Comment (datapythonista) [inline pandas/tseries/offsets.py]: I don't think this indentation is correct, is it?

**Review by datapythonista — COMMENTED** [review]
Just one thing that I'm not sure is correct, other than that looks great.

File: pandas/tseries/offsets.py (hunk 35 above)
Comment (varadgunjal) [reply pandas/tseries/offsets.py]: No it isn't. My bad.

**Review by datapythonista — APPROVED** [review]
Looks great @varadgunjal, thanks a lot for all the fixes.

@pandas-dev/pandas-core can someone take a look at this? It's just fixes to the docstrings formats, but it's plenty of them, and would be great to get this merged before it starts to have conflicts.

File: pandas/core/frame.py [hunk 36]
```diff
@@ -1579,7 +1590,8 @@ def to_records(self, index=True, convert_datetime64=None):
 
     @classmethod
     def from_items(cls, items, columns=None, orient='columns'):
-        """Construct a dataframe from a list of tuples
+        """
+        Construct a dataframe from a list of tuples.
```
Comment (gfyoung) [inline pandas/core/frame.py]: dataframe --> DataFrame

File: pandas/core/frame.py [hunk 37]
```diff
@@ -2887,7 +2895,8 @@ def _getitem_frame(self, key):
         return self.where(key)
 
     def query(self, expr, inplace=False, **kwargs):
-        """Query the columns of a frame with a boolean expression.
+        """
+        Query the columns of a frame with a boolean expression.
```
Comment (gfyoung) [inline pandas/core/frame.py]: frame -- >DataFrame

File: pandas/io/json/normalize.py [hunk 38]
```diff
@@ -12,7 +12,9 @@
 
 
 def _convert_to_line_delimits(s):
-    """Helper function that converts json lists to line delimited json."""
+    """
+    Helper function that converts json lists to line delimited json.
```
Comment (gfyoung) [inline pandas/io/json/normalize.py]: json --> JSON

**Review by gfyoung — APPROVED** [review]

**Comment by datapythonista** [comment]
Thanks @varadgunjal for all the work with this, and thanks @gfyoung for the review.
